server/core_leaderboard.go

```diff
@@ -968,3 +974,47 @@ func calculateExpiryOverride(overrideExpiry int64, leaderboard *Leaderboard) (in
 	}
 	return overrideExpiry, true
 }
+
+func disableLeaderboardRanks(ctx context.Context, logger *zap.Logger, db *sql.DB, leaderboardCache LeaderboardCache, rankCache LeaderboardRankCache, id string) error {
+	l := leaderboardCache.Get(id)
+	if l == nil || l.IsTournament() {
+		return runtime.ErrLeaderboardNotFound
+	}
+
+	if _, err := db.QueryContext(ctx, "UPDATE leaderboard SET enable_ranks = false WHERE id = $1", id); err != nil {
+		logger.Error("failed to set leaderboard enable_ranks value", zap.Error(err))
+		return errors.New("failed to disable tournament ranks")
+	}
+
+	l.EnableRanks = false
```

Comment:
Is this a safe operation, or is there a chance we'd get a panic if `l.EnableRanks` is read concurrently for this leaderboard?